cpp/CMakeLists.txt

```diff
@@ -60,6 +60,15 @@ rapids_find_package(
 )
 if(NOT cuFile_FOUND)
   message(WARNING "Building KvikIO without cuFile")
+else()
+  file(READ "${cuFile_INCLUDE_DIRS}/cufile.h" CUFILE_H_STR)
+  string(FIND "${CUFILE_H_STR}" "cuFileBatchIOSetUp" cuFileBatchIOSetUp_location)
+  if(cuFileBatchIOSetUp_location EQUAL "-1")
+    set(cuFile_BATCH_API_FOUND FALSE)
+  else()
+    set(cuFile_BATCH_API_FOUND TRUE)
+  endif()
+  message(STATUS "Found cuFile's Batch API: ${cuFile_BATCH_API_FOUND}")
```

Comment:
The right thing to do is to attempt to compile and link a call to this function, surely this is something that cmake supports? To ensure that both the header file and the library actually offer the function. 

This is fine, I guess, but there must be a canonical way.

Reply:
`check_symbol_exists` will try to compile and link a small executable that verifies that symbol exists. For this to work properly for libcufile you would also need to tell `check_symbol_exists` all the extra include directories, and link requirements.

The examples in the docs are easy since they are testing for values found in headers provided by the compiler/runtime and therefore no extra information is required.

Using CMake's `try_compile` would be easier in comparison but would still require setting up a code snippet and specifying the include directory for the libcufile and cuda rutime headers